### PR TITLE
[AUTOTVM] Allow fallback for template & Fix bugs in tuners

### PR DESCRIPTION
**Pull request by merrymercy**
Previous dispatchers do not allow fallback, because we want to make sure of the performance.
But topi call is widely used and we should enable fallback by default.

This RP
* adds fallback support for all templates. This is done by introducing a root fallback dispatcher.
* clean up some topi tests
* fix bugs in xgboost tuner, ga tuner
    * more exploration in xgb tuner
    * fix multiprocessing issues in xgb tuner
    * fix corner cases in ga tuner

### CODE REVIEW

**Comment by tqchen** [comment]
please request reviews

**Comment by merrymercy** [comment]
@tqchen @eqy pls review

File: topi/tests/python/test_topi_conv2d.py [hunk 1]
```diff
@@ -1,47 +0,0 @@
-"""Example code to do conv2d."""
```
Comment (eqy) [inline topi/tests/python/test_topi_conv2d.py]: Why do we remove this test?

File: topi/tests/python/test_topi_conv2d_nchw.py [hunk 2]
```diff
@@ -1,12 +1,14 @@
 """Example code to do convolution."""
```
Comment (eqy) [inline topi/tests/python/test_topi_conv2d_nchw.py]: Does this only use workloads that exist in current popular networks e.g., (ResNet-50, VGG, Superresolution)? If so, we may just wan to add some weird shapes to test the fallback mechanism.

Comment (merrymercy) [reply topi/tests/python/test_topi_conv2d_nchw.py]: I don't load pre-tuned parameters in this test. All these workloads will use fallback configs.

**Comment by merrymercy** [comment]
@tqchen  it is ready for merging, can you review?

File: topi/tests/python/test_topi_conv2d_nchw.py [hunk 3]
```diff
@@ -38,66 +48,103 @@ def check_device(device):
...
+    verify_conv2d_nchw(1,   48,  35,  64, 5, 1, 2)
+    verify_conv2d_nchw(1,   64,  35,  96, 3, 1, 1)
+    verify_conv2d_nchw(1,   96,  35,  96, 3, 1, 1)
+    verify_conv2d_nchw(1,  192,  35,  32, 1, 1, 0)
+    verify_conv2d_nchw(1,  256,  35,  64, 1, 1, 0)
+    verify_conv2d_nchw(1,  256,  35,  48, 1, 1, 0)
+    verify_conv2d_nchw(1,  288,  35,  64, 1, 1, 0)
+    verify_conv2d_nchw(1,  288,  35,  48, 1, 1, 0)
+    verify_conv2d_nchw(1,  288,  35, 384, 3, 2, 0)
+    # verify_conv2d_nchw(1,   96,  35,  96, 3, 2, 0)
```
Comment (merrymercy) [inline topi/tests/python/test_topi_conv2d_nchw.py]: disable for now due to the buts in cuda schedule

**Comment by merrymercy** [comment]
@FrozenGene I updated the `plan_size` of XGBTuner from 32 to 64. It can make the xgb tuner run faster. I don't observer performance regression in tuning.

**Comment by FrozenGene** [comment]
@merrymercy Thanks for information. And I am also happy to see that we solve the memory leak problem, which I planned to PR it (Sorry for the slow slow slow open source process of my company). My way is add __del__ in xgboost_cost_model.py. BTW, change plan_size from 32 to 64, how much we can improver?
